documentation/getstarted/advanced/migrationtesting.md

```diff
@@ -0,0 +1,241 @@
+
+    echo "Successfully migrated 'Pagila' database"
+
+    spawnctl delete data-container $pagilaContainerName --accessToken $SPAWNCTL_ACCESS_TOKEN -q
+
+    echo "Successfully cleaned up the Spawn data container '$pagilaContainerName'"
+    ```    
+
+1. Before committing and pushing this file to the repository, lets give it executable permissions:
+
+    <pre class="console"><span>&gt;</span> chmod +x migration-test-prod.sh</pre>
+
+This script accomplishes a few things. We are:
+  * Installing Spawn to give us access to the `spawnctl` command from the pipeline agent
+  * Creating a [Spawn data container](https://www.spawn.cc/docs/concepts-data-container) from the data image we created locally named `Pagila:prod`
+  * Using the official [Flyway docker image](https://hub.docker.com/r/flyway/flyway) to run `flyway migrate` on our data container, using migration scripts stored under the `sql` folder
+  * Automating the cleanup of the database
+
+That it - that is our migration test. We have quickly provisioned a database instance from our back up using Spawn, and set the flyway connection details to point to that database and run the migration scripts in our repository. Any errors will be apparent here and show up on the developers Pull Request that this ran against.
+
+Note: There is a [Flyway Migration action](https://github.com/marketplace/actions/flyway-migration) in the GitHub Marketplace which you can copy from. But using the code in `migrate-test.sh` is using generic bash and will work across all CI pipelines.
+
+### Run Flyway migration tests
+
+Once you've pushed all your changes to GitHub, you can now manually run the migration test workflow by navigating to the Actions tab in GitHub and clicking on 'Database migration test', then 'Run workflow'. Or, a more likely scenario, we want this to automatically run on a PR request so the development team can test that any new migrations will run against master before merging.
+
+Add the following to the top of your `migration-test.yml` file:
+
+```
+# Controls when the action will run. 
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch
+```
+
+### Automate creating backups 
+
+One last thing which is useful - with the way it's set up, you will have to create a data image manually every time you have migration changes to get the latest backup of your production database. But we can [automate this step](https://www.spawn.cc/docs/howto-ci-scheduled-image-creation) using Spawn in our existing GitHub Actions pipeline.
+
+1. Add some more GitHub secrets so that we can allow the agent access to our production database **to back up, not write to**. We will need the database's:
+  * `Host`
+  * `Username`
+  * `Password`
```

Comment:
It might be worth mentioning what the names of the secrets should be here

Reply:
f0b2c5f